win32ss/user/ntuser/ghost.c

```diff
@@ -128,6 +219,109 @@ HWND FASTCALL UserHungWindowFromGhostWindow(HWND hwndGhost)
     return IntHungWindowFromGhostWindow(pGhostWnd);
 }
 
+static BOOL IntHaveToQuitGhosting(void)
+{
+    return !!NtUserFindWindowEx(NULL, NULL, &GhostClass, NULL, 0);
+}
+
+static ULONG NTAPI
+GhostThreadProc(PVOID Param)
+{
+    HWND hwndTarget = (HWND)Param;
+    MSG msg;
+
+    if (!IntCreateGhostWindow(hwndTarget))
+    {
+        DPRINT1("Unable to create ghost\n");
+        goto Quit;
+    }
+
+    // thread message loop
+    while (NtUserGetMessage(&msg, NULL, 0, 0))
```

Comment:
This is going to fail from a system thread because `&msg` is a kernel pointer.

Reply:
NtUserGetMessage call is replaced with co_IntGetPeekMessage call.